time/timebase.c

```diff
@@ -50,10 +55,21 @@ void time_term() {
 
 void time_update(enum time_source source, enum time_source_event event, double event_arg) {
     struct timespec tv = {0, 0};
-    if (clock_gettime(CLOCK_MONOTONIC_RAW, &tv) != 0) {
-        PERROR("clock_gettime failed");
-        return;
-    }
+
+    #ifdef LINUX
+        if (clock_gettime(CLOCK_MONOTONIC_RAW, &tv) != 0) {
+            PERROR("clock_gettime failed");
+            return;
+        }
+    #elif OSX
+        clock_serv_t cclock;
+        mach_timespec_t mts;
+        host_get_clock_service(mach_host_self(), CALENDAR_CLOCK, &cclock);
```

Comment:
This is marked obsolete, not that it matters; the more common way to do this is mach_absolute_time.  Note that that clock doesn't move during device sleep, which may or may not be what you want; they apparently added mach_continuous_time in 10.12 for a clock that does move in sleep.